@blackhu thanks for the PR! I've adjusted the error message accordingly. If an image is not found it seems to be caught earlier already:

```
yolo predict source=fake.jpg
WARNING ⚠️ 'model' argument is missing. Using default 'model=yolov8n.pt'.
Ultralytics YOLOv8.2.36 🚀 Python-3.11.2 torch-2.3.0 CPU (Apple M2)
YOLOv8n summary (fused): 168 layers, 3151904 parameters, 0 gradients, 8.7 GFLOPs

Traceback (most recent call last):
  File "/Users/glennjocher/PycharmProjects/ultralytics/venv/bin/yolo", line 8, in <module>
    sys.exit(entrypoint())
             ^^^^^^^^^^^^
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/cfg/__init__.py", line 591, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/engine/model.py", line 453, in predict
    return self.predictor.predict_cli(source=source) if is_cli else self.predictor(source=source, stream=stream)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/engine/predictor.py", line 183, in predict_cli
    for _ in gen:  # sourcery skip: remove-empty-nested-block, noqa
  File "/Users/glennjocher/PycharmProjects/ultralytics/venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 35, in generator_context
    response = gen.send(None)
               ^^^^^^^^^^^^^^
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/engine/predictor.py", line 226, in stream_inference
    self.setup_source(source if source is not None else self.args.source)
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/engine/predictor.py", line 198, in setup_source
    self.dataset = load_inference_source(
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/data/build.py", line 202, in load_inference_source
    dataset = LoadImagesAndVideos(source, batch=batch, vid_stride=vid_stride)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/data/loaders.py", line 292, in __init__
    raise FileNotFoundError(f"{p} does not exist")
FileNotFoundError: fake.jpg does not exist
```

--- a/ultralytics/data/loaders.py
+++ b/ultralytics/data/loaders.py
@@ -362,7 +362,7 @@ class LoadImagesAndVideos:
                 self.mode = "image"
                 im0 = cv2.imread(path)  # BGR
                 if im0 is None:
-                    raise FileNotFoundError(f"Image Not Found or Unable to Open Error {path}")
+                    raise FileNotFoundError(f"Image Read Error {path}")
                 paths.append(path)
                 imgs.append(im0)
                 info.append(f"image {self.count + 1}/{self.nf} {path}: ")

